### DISCUSSION TITLE
0.4.0

### DISCUSSION THREAD

**Discussion by kubilus1** [kubilus1/autoortho#82] [Announcements]
Preview release.  Contains massive performance improvements.  Since this is a pre-release there may be bugs/visual artifacts, etc.

## What's Changed
* Mm0 partial + 2 bug fixes by @hotbso in https://github.com/kubilus1/autoortho/pull/54
* Integrate Aoimage from @hotbso by @kubilus1 in https://github.com/kubilus1/autoortho/pull/74
* Update winsetup.py by @marcfsx in https://github.com/kubilus1/autoortho/pull/79
* Fix when winfsp is not present by @kubilus1 in https://github.com/kubilus1/autoortho/pull/80
* Support partial compression of mipmap 0 by @kubilus1 in https://github.com/kubilus1/autoortho/pull/58

## New Contributors
* @hotbso made their first contribution in https://github.com/kubilus1/autoortho/pull/54
* @marcfsx made their first contribution in https://github.com/kubilus1/autoortho/pull/79

**Full Changelog**: https://github.com/kubilus1/autoortho/compare/0.3.1-4...0.4.0-rc0

<hr /><em>This discussion was created from the release <a href='https://github.com/kubilus1/autoortho/releases/tag/0.4.0-rc0'>0.4.0</a>.</em>

**Comment by tommypank** [comment]
Getting some pink / black artifacts on the ortho with this release?
![X-Plane Screenshot 2023 03 19 - 17 41 08 03](https://user-images.githubusercontent.com/76015940/226195972-045b5b4e-3707-4521-811d-5ed5528eae96.png)

**Reply by kubilus1** [reply]
XP version?  Windows ver?  Any errors in the logs?

**Reply by tommypank** [reply]
XP12.04 latest win 11 22621. No visible errors in the logs

**Reply by tommypank** [reply]
About 10 hours on the new version, amazing performance and the pink squares that seem to change upon angle of view are only within the first stage of flight when loading the initial tiles I have found.

**Comment by Awemeter** [comment]
Windows still flags it as a virus. I also decided to run it through virus total and about 4 sources flagged it as malicious. https://www.virustotal.com/gui/file/9a5d23d02f38c0e3a53d3718ba50dc39dd1502b9bee774ac43a604e368b8b478/detection.

**Reply by tommypank** [reply]
Mine has started flagging after about 4 days of use.

**Reply by Awemeter** [reply]
Update: Windows still blocks the download of Auto Ortho.

**Reply by Jeff2122** [reply]
As of this morning, the executable runs without issues with Defender still enabled (time will tell). But the Python run.bat version disappears when the server starts. EDIT: Scanned clean with Defender! Yay!

**Reply by Awemeter** [reply]
Just tried to download it and it is still being blocked.

**Reply by Jeff2122** [reply]
Wow, I don't understand what's going on. I just downloaded a new copy and ran it with no issues at all. Maybe MS takes time to get around to poking a new antivirus list out to the world?
This version is truly what I was hoping for. It loads in a very reasonable time now. Still pauses once in while to catch up, but it continues.

**Reply by Awemeter** [reply]
Still gets flagged. @kubilus1 is there nothing you can do on the code side to fix the false positives?

**Reply by Jeff2122** [reply]
Sorry to hear you are still being locked out. I have Win 10 here, located in New Mexico, USA, and it is no longer being flagged on my system. I did bypass the antivirus for a day or two but then tried it Tuesday and it worked without disabling AV. No magic. I hope you can get it going.

**Reply by Awemeter** [reply]
I am on windows 10 build 19044.2728 in the United States.

**Comment by mikecolive** [comment]
Is it 0.4.0 already available?

**Reply by kubilus1** [reply]
Early release.

**Reply by mikecolive** [reply]
Yeah. Just tried it now. Half or even a bit less of loading times now. Awesome

**Comment by pntrafewl** [comment]
Flew JFK-MIA with 0.4.0. Awesome performance and not one stutter.

**Comment by Jeff2122** [comment]
Downloaded an EXE yesterday and Windows AV killed it. Today I grabbed ortho-main, and the server window just disappears after starting it. Won't run. Please advise.

**Reply by Awemeter** [reply]
You can either disable the AV, allow autoortho manually, or wait until windows fixes the false positive. The dev already submitted it to windows defender.

**Reply by Jeff2122** [reply]
Yes, thank you, not sure how long I can fly as Windows says it will turn back on automatically. I tried the python version and it still disappears, even though the AV is disabled. But the .exe works. And wow, how much faster it loads. Impressive!

**Comment by pntrafewl** [comment]
Been doing some flying around south america and still not one stutter. I've never had such high frame rates in VR, even with the crappy default ground textures.

**Reply by kubilus1** [reply]
That's good to hear.

**Comment by evilstiefel** [comment]
Can I just say, this is one of the best addons for X-Plane scenery, and it gets better with every version. Thank you so much for your efforts! Looking forward to any future enhancements :).

**Reply by kubilus1** [reply]
Thanks!  Glad it's useful.  Got some more stuff in the works, but super busy ATM.

**Comment by dillan4561** [comment]
![FLIGHT](https://user-images.githubusercontent.com/35738129/227977476-75ce352c-ca56-4533-a153-47c690f9b965.png)

im getting this error constantly running through when autoortho is open, its causing stutters and high cpu spikes throughout the flight, any help?

**Reply by kubilus1** [reply]
Do you have a non standard XPlane UDP data port or Xplane data connections disabled?

**Reply by dillan4561** [reply]
where would that be, i have the stuff in network all enabled and ive got the "accept incoming connections turned on

**Reply by dillan4561** [reply]
![xp](https://user-images.githubusercontent.com/35738129/228395006-665b9bbf-e0c0-4f4c-8273-bb609b278673.png)

i get this error tho

**Comment by CptTai** [comment]
This is by far the best scenery enhancer for XP. Thank you so much for contributing to this community!

**Comment by NichoAir** [comment]
Using the .exe i get some python errors when i click run. I retried using the source and the run.bat and get this error message:

ERROR: Pillow_SIMD-9.0.0.post0-cp310-cp310-win_amd64.whl is not a supported wheel on this platform.

I assume the python errors i get when i use the .exe have something to do with this error

![error](https://user-images.githubusercontent.com/129181852/228528808-1575d263-9bbd-46bc-857e-024269108d62.PNG)

the old Version (0.3.1) is running without problems

**Reply by kubilus1** [reply]
Try using python3.10 instead of python3.11

**Reply by NichoAir** [reply]
Using python 3.10 helped the Error from the Text, bu I also found a solution for the error shown in the picture:
I added a "import sys" in the C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\refuse\_refactor.py
